src/server/protocol.ts

```diff
@@ -2308,9 +2316,14 @@ namespace ts.server.protocol {
         codeActions?: CodeAction[];
 
         /**
-         * Human-readable description of the `source` from the CompletionEntry.
+         * @deprecated Use `sourceDisplay` instead.
          */
         source?: SymbolDisplayPart[];
+
+        /**
+         * Human-readable description of the `source` from the CompletionEntry.
+         */
+         sourceDisplay?: SymbolDisplayPart[];

```

Comment:
Aiming to stop overloading the meaning of `source` between `CompletionEntry` and `CompletionEntryDetails`. In an ideal world, `source` is the machine-readable module name for the auto-import, and `sourceDisplay` is the human-readable module specifier for it, no matter whether you’re looking at `CompletionEntry` or `CompletionEntryDetails`. In reality, we will continue to set `CompletionEntryDetails["source"]` to the human-readable module specifier for backward compatibility, but I’ve marked it as deprecated.

Reply:
The editor only pulls on these one at a time right?